tests/validator/validator.cpp

```diff
@@ -1065,3 +1065,726 @@ TEST(Validator, validMathCnElements)
     v.validateModel(m);
     EXPECT_EQ(size_t(0), v.errorCount());
 }
+
+TEST(Validator, unitAmericanSpellingOfUnitsRemoved)
+{
+    libcellml::Validator validator;
+    libcellml::ModelPtr m = std::make_shared<libcellml::Model>();
+    libcellml::ComponentPtr comp1 = std::make_shared<libcellml::Component>();
+    libcellml::ComponentPtr comp2 = std::make_shared<libcellml::Component>();
+
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+
+    v1->setName("tomayto");
+    v2->setName("tomahto");
+    m->setName("callthewholethingoff");
+    comp1->addVariable(v1);
+    comp2->addVariable(v2);
+    comp1->setName("comp1");
+    comp2->setName("comp2");
+    m->addComponent(comp1);
+    m->addComponent(comp2);
+
+    // u1 = u2: different spelling of meter/metre
+    libcellml::UnitsPtr u1 = std::make_shared<libcellml::Units>();
+    u1->setName("testunit1");
+    u1->addUnit("metre");
+    libcellml::UnitsPtr u2 = std::make_shared<libcellml::Units>();
+    u2->setName("testunit2");
+    u2->addUnit("meter");
+
+    v1->setUnits(u1);
+    v2->setUnits(u2);
+    m->addUnits(u1);
+    m->addUnits(u2);
+
+    const std::vector<std::string> expectedErrors = {
+        "Units reference 'meter' in units 'testunit2' is not a valid reference to a local units or a standard unit type.",
+        "Variable 'tomayto' has units of 'testunit1' and an equivalent variable 'tomahto' with non-matching units of 'testunit2'. The mismatch is: metre^1, ",
+        "Variable 'tomahto' has units of 'testunit2' and an equivalent variable 'tomayto' with non-matching units of 'testunit1'. The mismatch is: metre^-1, "};
```

Comment:
The last two error messages are effectively the same. So, only one of them ought to be sufficient.

Also, the last two error messages end up with `, `, which is clearly not what we want. We want a properly formatted error message.

Reply:
Done :)